araproc/analysis/hilbert_envelope_snr.py

```diff
@@ -0,0 +1,110 @@
+# this calculates Hilbert envelope SNR for each channel and their average
+
+import numpy as np
+from scipy.signal import hilbert
+
+from araproc.analysis import snr
+from araproc.framework import waveform_utilities as wfu
+
+
+
+def get_envelope(waveform, nsegs = 8):
+
+    """
+    Calculates the Hilbert envelope voltage and rms of a trace.
+
+    Parameters
+    ----------
+    waveform: TGraph
+        A TGraph of the waveform.
+    nsegs: int
+        Number of segments to break waveform into.
+
+    Returns
+    -------
+    hill_max: float
+        Hilbert envelope maximum voltage in same units as trace.
+    hill_rms: float
+        Hilbert envelope minimum rms among all segments in same units as trace..
+    """ 
+
+    _, trace = wfu.tgraph_to_arrays(waveform)
+   
+    hill = np.abs(hilbert(trace))
+    hill_max_idx = np.nanargmax(hill)
+    hill_max = hill[hill_max_idx]
+  
+    ''' 
+    segments = np.array_split(hill, nsegs)
+    rms_segments = [np.sqrt(np.nanmean(segment**2)) for segment in segments]
+    sorted_rms_segments = sorted(rms_segments)
+    hill_rms = np.mean(sorted_rms_segments[:3])    
+    '''
+
+    hill_rms = snr.get_min_segmented_rms(hill)
+ 
+    return hill_max, hill_rms
+
+
+
+
+def get_hill_snr(waveform):
+
+    """
+    Calculates Hilbert envelope SNR of a single voltage trace.
+
+    Parameters
+    ----------
+    waveform: TGraph
+        A TGraph of the waveform.
+
+    Returns
+    -------
+    hill_snr : float
+        The Hilbert envelope SNR of the waveform.
+    """
+
+    hill_max, hill_rms = get_envelope(waveform)
```

Comment:
do we need a separate `get_envelope` function? Seems like this is simple enough to put into one function.

Reply:
I think a `get_envelope` function would make sense if it returns the Hilbert envelope itself. With that we could easily do what you are saying and also have the benefit of being able to, eg, plot it. Otherwise, I think it can be merged.